### PR TITLE
[thermalctld] fix some redundant removal of state DB tables

### PR DESCRIPTION
**Pull request by vdahiya12**
Signed-off-by: vaibhav-dahiya <vdahiya@microsoft.com>
This PR aims to get rid of such messages in syslog when thermalctld, essentially delete keys is getting called twice, causing some redis issues.
This PR gets rid of deleting such keys twice.

Nov 16 00:33:00.566174 sonic INFO pmon#/supervisord: thermalctld Exception ignored in: <function TemperatureUpdater.__del__ at 0xb5dbe810>
Nov 16 00:33:00.566585 sonic INFO pmon#/supervisord: thermalctld Traceback (most recent call last):
Nov 16 00:33:00.566825 sonic INFO pmon#/supervisord: thermalctld   File "/usr/local/bin/thermalctld", line 558, in __del__
Nov 16 00:33:00.567014 sonic INFO pmon#/supervisord: thermalctld     table_keys = self.table.getKeys()
Nov 16 00:33:00.567181 sonic INFO pmon#/supervisord: thermalctld   File "/usr/lib/python3/dist-packages/swsscommon/swsscommon.py", line 2320, in getKeys
Nov 16 00:33:00.567343 sonic INFO pmon#/supervisord: thermalctld     return _swsscommon.Table_getKeys(self)
Nov 16 00:33:00.567499 sonic INFO pmon#/supervisord: thermalctld RuntimeError: RedisReply catches system_error: command: *2#015
Nov 16 00:33:00.567658 sonic INFO pmon#/supervisord: thermalctld $4#015
Nov 16 00:33:00.567811 sonic INFO pmon#/supervisord: thermalctld KEYS#015
Nov 16 00:33:00.567964 sonic INFO pmon#/supervisord: thermalctld $18#015
Nov 16 00:33:00.568118 sonic INFO pmon#/supervisord: thermalctld TEMPERATURE_INFO|*#015
Nov 16 00:33:00.568273 sonicINFO pmon#/supervisord: thermalctld , reason: Expected to get redis type 2 got type 3, err: NON-STRING-REPLY: Input/output error: Input/output error

### CODE REVIEW

**Comment by prgeor** [comment]
@vdahiya12 could you update
```
Description
Motivation and Context
How Has This Been Tested?
```

File: sonic-thermalctld/scripts/thermalctld [hunk 1]
```diff
@@ -565,16 +565,8 @@ class TemperatureUpdater(logger.Logger):
...
-    def deinit(self):
-        """
-        Deinitializer of TemperatureUpdater
-        :return:
-        """
-        for name in self.temperature_status_dict.keys():
-            self.table._del(name)
-            if self.is_chassis_system and self.chassis_table is not None:
-                self.chassis_table._del(name)
+        if self.is_chassis_system and self.chassis_table is not None:
```
Comment (prgeor) [inline sonic-thermalctld/scripts/thermalctld:568]: this issue is not seen on fixed platform i.e non-modular chassis?

File: sonic-thermalctld/scripts/thermalctld [hunk 2]
```diff
@@ -565,16 +565,8 @@ class TemperatureUpdater(logger.Logger):
...
+            self.chassis_table._del(name)
```
Comment (prgeor) [inline sonic-thermalctld/scripts/thermalctld:569]: is this tested?

**Comment by Junchao-Mellanox** [comment]
Please add new unit test case and fix the existing test case failure.

File: sonic-thermalctld/scripts/thermalctld (hunk 2 above)
Comment (Junchao-Mellanox) [inline sonic-thermalctld/scripts/thermalctld:569]: where is name defined?

**Review by Junchao-Mellanox — CHANGES_REQUESTED** [review]

File: sonic-thermalctld/scripts/thermalctld (hunk 1 above)
Comment (vdahiya12) [reply sonic-thermalctld/scripts/thermalctld]: not sure, need to investgate

File: sonic-thermalctld/scripts/thermalctld (hunk 2 above)
Comment (vdahiya12) [reply sonic-thermalctld/scripts/thermalctld]: fixed

Comment (vdahiya12) [reply sonic-thermalctld/scripts/thermalctld]: yes, tested on M0 failing device

**Comment by prgeor** [comment]
@Junchao-Mellanox any more comments.? LGTM

**Review by Junchao-Mellanox — APPROVED** [review]

**Review by prgeor — APPROVED** [review]

**Pull request merged**